src/hooks/useGasLimit.ts

```diff
@@ -8,20 +8,21 @@ import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
 import useSafeAddress from './useSafeAddress'
 import useWallet from './wallets/useWallet'
 import { useSafeSDK } from './coreSDK/safeCoreSDK'
+import useIsSafeOwner from './useIsSafeOwner'
 
 export const getPreValidatedSignature = (from: string): string => {
   return `0x000000000000000000000000${from
     .toLowerCase()
     .replace('0x', '')}000000000000000000000000000000000000000000000000000000000000000001`
 }
 
-export const _encodeSignatures = (safeTx: SafeTransaction, from: string): string => {
-  const owner = from.toLowerCase()
-  const needsOwnerSig = !safeTx.signatures.has(owner)
+export const _encodeSignatures = (safeTx: SafeTransaction, from?: string): string => {
+  const owner = from?.toLowerCase()
+  const needsOwnerSig = owner && !safeTx.signatures.has(owner)
 
   // https://docs.gnosis.io/safe/docs/contracts_signatures/#pre-validated-signatures
   if (needsOwnerSig) {
-    const ownerSig = getPreValidatedSignature(from)
+    const ownerSig = getPreValidatedSignature(owner)
```

Comment:
You can use `generatePreValidatedSignature` from the `safe-core-sdk` here. We then don't even need `getPreValidatedSignature` in the app, i.e.

```ts
if (needsOwnerSig) {
  const signature = generatePreValidatedSignature(owner)
  safeTx.addSignature(signature)
}
```

Reply:
Good call, didn't know this fn existed!